course/templates/course/index.html

```diff
@@ -12,85 +12,148 @@ <h1>{{ welcome_text|safe }}</h1>
 </div>
 {% endif %}
 
-<div class="row course-thumbnails">
-    {% site_advert as advert %}
-    {% if advert %}
+{% if my_instances %}
+<div class="frontpage-panel">
+    <h2 class="panel-heading">{% trans "My courses" %}</h2>
+    <div class="row frontpage">
+        {% for instance in my_instances %}
+        {% with instance_url=instance|url %}
         <div class="col-sm-6 col-md-4 col-lg-3">
-            <div class="thumbnail advert-thumbnail">
-                <div class="image" aria-hidden="true">
-                    {% if advert.image %}
-                        <img src="{{ advert.image }}" alt="">
-                    {% else %}
-                        <h1 class="glyphicon glyphicon-comment"></h1>
-                    {% endif %}
+            <div class="frontpage-course card">
+                <a href="{{ instance_url }}">
+                    <div class="course-image" aria-hidden="true">
+                        {% if instance.image %}
+                        <img class="card-img-top" src="{{ instance.image.url }}" alt="" />
+                        {% else %}
+                        <div class="card-img-top">
+                            <i class="glyphicon glyphicon-book"></i>
+                        </div>
+                        {% endif %}
+                    </div>
+                </a>
+                <div class="card-body">
+                    <h3 class="card-title" lang="{{ instance.language }}">
+                        <a href="{{ instance_url }}">
+                            {{ instance.course.name|parse_localization }}<br />
+                        </a>
+                    </h3>
+                    <h4 class="card-subtitle" lang="{{ instance.language }}">
+                        {{ instance.instance_name|parse_localization }}
+                    </h4>
+                    <p class="card-text">
+                        {{ instance.course.code }}
+                        {% if not instance.visible_to_students %}
+                        <span class="label label-danger">{% trans "Hidden from students" %}</span>
+                        {% endif %}
+                        <br />
+                        {{ instance.starting_time|date:"SHORT_DATE_FORMAT" }} &ndash;
+                        {{ instance.ending_time|date:"SHORT_DATE_FORMAT" }}
+                        <br />
+                        {% if instance.enrollment_audience == 1 or instance.enrollment_audience == 3 %}
+                        <span class="label label-success">{{ internal_user_label|safe }}</span>
```

Comment:
Do we benefit from showing a label on courses for "home institution" students? That's the default, I'd assume, and we could clean up the block a lot by skipping it. 

It's likely that this is also out of scope for this issue, but definitely worth considering in the future.

Reply:
Good discussion. Ideally there could be courses that are only available to the external users or such. In addition, some courses might not be open for anyone from aalto, e.g., if they have manua enrollment done by the teacher. Thus, I would wish there would be filters that could be used to limit what is shown. These labels are currently kind of placeholders for that kind of "who can do the course" information. Actually desigining and implementing that would be cool.

Reply:
Definitely, I think that was a good point Matthew made. I could continue with the filters and search once I get this issue wrapped up. I will create an issue on that, after this one closes, and I would like both of your input on what those filters, searches and badges should do so that I can implement them properly. 

From the point of this issue and PR, I think we can skip this now, and continue with this work in the next issue.

Reply:
> From the point of this issue and PR, I think we can skip this now, and continue with this work in the next issue.

Yes. I think that is good. Let's continue the discussion in a dedicated issue.

Reply:
I made an issue for this:
https://github.com/apluslms/a-plus/issues/580

And few other related issues: 
https://github.com/apluslms/a-plus/issues/579
https://github.com/apluslms/a-plus/issues/578